doc/weakly_supervised.rst

```diff
@@ -700,6 +700,60 @@ of triplets that have the right predicted ordering.
 Algorithms
 ----------
 
+.. _scml:
+
+:py:class:`SCML <metric_learn.SCML>`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Sparse Compositional Metric Learning
+(:py:class:`SCML <metric_learn.SCML>`)
+
+`SCML` learns an squared mahalanobis distance from triplet constraints by 
+optimizing sparse positive weights assigned to a set of :math:`K` locally discriminative 
+rank-one PSD bases. This can be formulated as an optimization problem with only :math:`K`
+parameters, that can be solved with an efficient stochastic composite scheme.
+
+The Mahalanobis Matrix :math:`M` is built from a basis set :math:`B = \{b_i\}_{i=\{1,...,K\}}`
+weighted by a :math:`K` dimensional vector :math:`w = \{w_i\}_{i=\{1,...,K\}}` as:
+
+.. math::
+
+    M = \sum_{i=1}^K w_i b_i b_i^T = B \cdot diag(w) \cdot B^T \quad w_i \geq 0
+
+Learning :math:`M` in this form makes it PSD by design, as it is a nonnegative sum of PSD matrices.
```

Comment:
Before "The optimization problem", add a sentence to clarify that the set of bases B is fixed in advance (typically generated from the input data).

Reply:
Good idea! I was aware I was missing mentioning that somewhere